packages/checkbox/src/Checkbox/Checkbox.tsx

```diff
@@ -50,6 +50,7 @@ function Checkbox({
   id: idProp,
   indeterminate: indeterminateProp,
   label = '',
+  defaultChecked = false,
```

Comment:
I believe other components use `initialValue` (e.g. Combobox). Unsure whether consistency with other LG components or with HTML checkbox api is more valuable

Reply:
The `Text Area` uses [defaultValue](https://github.com/mongodb/leafygreen-ui/blob/main/packages/text-area/src/TextArea/TextArea.tsx#L64). I would vote to use `defaultValue` across all components to meet HTML naming rule.

Reply:
I don't have a strong opinion; I'll defer to @TheSonOfThomp since you regularly work in this repo.